framework/src/media/InputHandler.cpp

```diff
@@ -25,29 +25,21 @@
 #include "MediaPlayerImpl.h"
 #include "Decoder.h"
 
-#ifndef CONFIG_HANDLER_STREAM_BUFFER_SIZE
-#define CONFIG_HANDLER_STREAM_BUFFER_SIZE 4096
-#endif
-
-#ifndef CONFIG_HANDLER_STREAM_BUFFER_THRESHOLD
-#define CONFIG_HANDLER_STREAM_BUFFER_THRESHOLD 2048
-#endif
-
 namespace media {
 namespace stream {
 
 InputHandler::InputHandler() :
-	mInputDataSource(nullptr),
 	mDecoder(nullptr),
-	mWorker(0),
-	mIsWorkerAlive(false),
 	mState(BUFFER_STATE_EMPTY),
 	mTotalBytes(0)
 {
+	mWorkerName = "InputHandler";
+	mWorkerStackSize = CONFIG_INPUT_DATASOURCE_STACKSIZE;
 }
 
 void InputHandler::setInputDataSource(std::shared_ptr<InputDataSource> source)
 {
+	StreamHandler::setDataSource(source);
```

Comment:
It is main object, validation must be checked.

Reply:
Fixed it